### ISSUE TITLE
main driver 2 - spatial occurrence

### ISSUE THREAD

**Issue by tmesaglio** [tmesaglio/Australian-plant-photos#19]
I still need to do a bit of minor cleaning of points randomly in the middle of the ocean, but here's the heat map (with thanks to Sophie for her awesome code). 

![image](https://user-images.githubusercontent.com/48739968/164679089-43c5c28e-d614-4ffd-853a-93dfc4899a12.png)

Looks pretty good!! Three hotspots = West Arnhem Land, SW Western Australia, and the Daintree

**Comment by tmesaglio** [comment]
and here's the same map, but at a resolution of 0.5 degrees x 0.5 degrees cells

![image](https://user-images.githubusercontent.com/48739968/164679626-5b524bbf-12f7-49b2-9f1b-b19945c0ac70.png)

**Comment by wcornwell** [comment]
maybe try this: https://www.rdocumentation.org/packages/CoordinateCleaner/versions/2.0-20/topics/cc_sea ?

**Comment by tmesaglio** [comment]
> maybe try this: https://www.rdocumentation.org/packages/CoordinateCleaner/versions/2.0-20/topics/cc_sea ?

yeah I tried that (I used it to clean a few records that had coordinates in herbaria), but unfortunately removing sea records also removes all of the island records; essentially it only treats continential Australia + Tasmania as 'land', and everything else like Xmas, Norfolk, Lord Howe etc get wiped out too. 

Should be pretty quick to fix manually

**Comment by wcornwell** [comment]
also, i'm usually not in favor of this, but this might be a good situation to split the continuous variable into bins e.g. 

1. 0 unphotographed species known
2. 1 unphotographed species known
3. 2-5 
4. 5-10
5. 10-100
6. >100

**Comment by wcornwell** [comment]
idk, might have to fiddle with the bins, but i think the difference between 0 and 1 and 1 and 10 are getting lost currently.

**Comment by tmesaglio** [comment]
@wcornwell new version with binned data (ignore the legend formatting, I have to fix that)
![image](https://user-images.githubusercontent.com/48739968/165516137-bdede7fc-1d71-4396-b7a9-f2b228c3a192.png)

**Comment by wcornwell** [comment]
Looking good! I think there are some new tricks with 'geom_raster' for discrete data to get the legend right. Will google...

**Comment by wcornwell** [comment]
maybe something like: https://r-graph-gallery.com/267-reorder-a-variable-in-ggplot2.html

**Comment by tmesaglio** [comment]
> @tmesaglio are you recovering from the bioblitz?
> 
> where is the code for this fig?

BioBlitz was fantastic, lots to tell you about 

the code for that figure should be in the R file ‘driver 1 - spatial analysis’, and it’s from line 124 onwards

**Comment by wcornwell** [comment]
![bettercolor](https://user-images.githubusercontent.com/5534161/167703013-954c91be-a0c7-4a9d-b576-0064ac73bafe.png)

**Comment by tmesaglio** [comment]
that's very nice, very clean looking. Thoughts on the legend being reversed? Highest number at top?

**Comment by wcornwell** [comment]
![rev_legend](https://user-images.githubusercontent.com/5534161/167754208-ae0cef60-5d11-4568-a71a-d84f3d77a480.png)

**Comment by tmesaglio** [comment]
beautiful!

**Comment by tmesaglio** [comment]
@wcornwell did you have the updated code for this polished version?

**Comment by wcornwell** [comment]
updated code here: https://github.com/tmesaglio/Australian-plant-photos/commit/5a5923f2d6bd4f543b3e429ba1592fada6ca8ad1
